app/controllers/TaxYearSummaryController.scala

```diff
@@ -193,10 +225,30 @@ class TaxYearSummaryController @Inject()(taxYearSummaryView: TaxYearSummary,
     }
   }
 
-  def taxYearsUrl(origin: Option[String]): String = controllers.routes.TaxYearsController.showTaxYears(origin).url
+  def renderAgentTaxYearSummaryPage(taxYear: Int): Action[AnyContent] = Authenticated.async { implicit request =>
+    implicit user =>
+      getMtdItUserWithIncomeSources(incomeSourceDetailsService, useCache = false) flatMap { implicit mtdItUser =>
+        withTaxYearFinancials(taxYear, true) { documentDetailsWithDueDates =>
```

Comment:
might be nice to have a 'handleRequest' func to remove the duplicated code here

Reply:
Agreed. Made changes.